### PR TITLE
fix: performance issues

### PR DESCRIPTION
**Pull request by six7**
Fixes some performance issues:

- Removes no longer available nodes from cache when launching
- Catches a case where the plugin would be stuck when erroring in the worker due to a node no longer existing
- Checks if nodes still exist at the time when we try to get node plugin data
- Fixes an error with local styles that caused it to fetch Figma styles leading to the worker being stuck sometimes

### CODE REVIEW

File: src/plugin/NodeManager.ts [hunk 1]
```diff
@@ -54,6 +54,27 @@ export class NodeManager {
...
+        const allNodes = figma.root.findAll();
+        if (allNodes.length > 0) {
+          const nodeIds = new Set(allNodes.map((node) => node.id));
+          const deletedNodeIds = new Set<string>();
+
+          // Remove any nodes from the cache that no longer exist
+          for (const [id] of this.persistentNodesCache) {
+            if (!nodeIds.has(id)) {
+              this.persistentNodesCache.delete(id);
+              deletedNodeIds.add(id);
```
Comment (SorsOps) [inline src/plugin/NodeManager.ts:67]: Do we need to track the deletedNodes , can't this just be a boolean?

**Review by SorsOps — COMMENTED** [review]
Just minor questions

File: src/plugin/NodeManager.ts [hunk 2]
```diff
@@ -54,6 +54,27 @@ export class NodeManager {
...
+            }
+          }
+
+          // Emit a cache update if we removed any nodes
+          if (deletedNodeIds.size > 0) {
+            const remainingEntries = Array.from(this.persistentNodesCache.entries()).filter(([id]) => !deletedNodeIds.has(id));
```
[the remark this answers is not in the document]
Comment (six7) [reply src/plugin/NodeManager.ts]: you're absolutely right, fixed.

File: src/plugin/NodeManager.ts [hunk 3]
```diff
@@ -203,7 +224,11 @@ export class NodeManager {
         promises.add(defaultWorker.schedule(async () => {
           const node = nodes[nodeIndex];
 
-          await this.getNodePluginData(node);
+          try {
+            await this.getNodePluginData(node);
```
[the remark this answers is not in the document]
Comment (six7) [reply src/plugin/NodeManager.ts:228]: agree, we should do that in a followup pr

File: src/plugin/NodeManager.ts (hunk 1 above)
Comment (six7) [reply src/plugin/NodeManager.ts]: changed to a boolean 👍

**Pull request closed without merging**